src/solution.c

```diff
@@ -468,12 +471,35 @@ static void solution_thread(void *arg)
       continue;
     }
 
+    /* check if we have a solution, if yes calc iono correction */
+    if (soln_flag) {
+      for (u8 i = 0; i < n_ready_tdcp; i++) {
+        ionosphere_t iono_params;
+        double az, el, iono_correction;
+        /* read iono parameters for the tracking channel if available */
+        if (gps_iono_params_read(&iono_params)) {
+          wgsecef2azel(nav_meas[i].sat_pos, position_solution.pos_ecef,
+                       &az, &el);
+          /* calculate iono correction */
+          iono_correction = calc_ionosphere(&nav_meas_tdcp[i].tot,
+                                            position_solution.pos_llh[0],
+                                            position_solution.pos_llh[1],
+                                            az,el,
+                                            &iono_params);
+          /* correct pseudorange */
+          nav_meas_tdcp[i].pseudorange -= iono_correction;
```

Comment:
After https://github.com/swift-nav/piksi_firmware/pull/651 lands (later today), you should also add a opposite sign correction for the carrier phase.